### PR TITLE
Renderjobs

### PR DESCRIPTION
**Pull request by majakudlicka**
Changes as per commit messages. Generally speaking, we now have a data from the database being displayed on the SearchJob page, and the functionality to query the database for specific job category.

### CODE REVIEW

File: database/database_queries.js [hunk 1]
```diff
@@ -18,6 +18,19 @@ data.getJobs = (term, callback) => {
     );
 };
 
+data.getRandomJobs = callback => {
+    dbConnection.query(
+        `SELECT * FROM jobs WHERE end_date < NOW()
+         ORDER BY start_date LIMIT 10`,
+        (err, res) => {
```
Comment (des-des) [inline database/database_queries.js]: This function does nothing, pass your callback in directly as the last argument to `connection.query`

Comment (des-des) [inline database/database_queries.js]: Then you can get the rows out in the handler

File: src/client/components/pages/SearchJobs.js [hunk 2]
```diff
@@ -1,19 +1,167 @@
...
 class BrowseJobs extends Component {
-    render() {
+    constructor() {
+        super();
+        this.renderJobs = this.renderJobs.bind(this);
+        this.formatDate = this.formatDate.bind(this);
+        this.formatTime = this.formatTime.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+        this._onChange = this._onChange.bind(this);
+        this.state = {
```
Comment (des-des) [inline src/client/components/pages/SearchJobs.js]: No, use redux for this!

File: src/client/components/pages/SearchJobs.js [hunk 3]
```diff
@@ -1,19 +1,167 @@
...
+        return date.slice(0, 10);
+    }
+
+    formatTime(time) {
+        return time.slice(0, 5);
+    }
+
+    _onSubmit(evt) {
+        evt.preventDefault();
+        Axios.get('/api/jobs', {
```
Comment (des-des) [inline src/client/components/pages/SearchJobs.js]: This should be handled with a redux action (checkout redux thunk)

File: src/server/server.js [hunk 4]
```diff
@@ -54,27 +55,50 @@ server.register(plugins, (err) => {
...
         method: 'GET',
-        path: '/info',
+        path: '/api/jobs',
         handler: (request, reply) => {
-            reply({
-                name: pkg.name,
-                version: pkg.version,
-                message: 'Welcome to BEEVR!'
+            data.getJobs(request.url.query.term, (err, res) => {
+                if (err)
```
Comment (des-des) [inline src/server/server.js]:
```js
if (err) {
  //log some stuff if you want
  reply.status(500)('error message for client') // some thing like this
} else {
  //reply normally
}
```

File: src/server/server.js [hunk 5]
```diff
@@ -54,27 +55,50 @@ server.register(plugins, (err) => {
...
         }
     });
-    
+
+    server.route({
+        method: 'GET',
+        path: '/api/random_jobs',
+        handler: (request, reply) => {
+            data.getRandomJobs((err, res) => {
+                if (err)
```
Comment (des-des) [inline src/server/server.js]: same as above!

Comment (majakudlicka) [inline src/server/server.js]: Done

File: src/server/server.js (hunk 4 above)
Comment (majakudlicka) [inline src/server/server.js]: Done

File: database/database_queries.js (hunk 1 above)
Comment (majakudlicka) [inline database/database_queries.js]: I dont understand :(

**Comment by des-des** [comment]
There are no new commits on this branch.

image

**Comment by des-des** [comment]
@majakudlicka what does done mean? there are no new commits on this branch

File: src/client/components/pages/SearchJobs.js (hunk 2 above)
Comment (majakudlicka) [inline src/client/components/pages/SearchJobs.js]: I wired up Redux.
FormatTime function simply helps with rendering, I changed two others but can't see how I can get rid of the binding statements

File: src/client/components/pages/SearchJobs.js (hunk 3 above)
Comment (majakudlicka) [inline src/client/components/pages/SearchJobs.js]: Done

**Comment by majakudlicka** [comment]
Updated to use Redux!

File: src/client/actions/search_jobs.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            dispatch({
+                type: FETCH_STUDENT_JOBS,
+                status: 'error',
+                error: err
+            });
+        });
+};
+
+export const fetchSelectedJobs = term => dispatch => {
+    console.log('fetch jobs called');
```
Comment (des-des) [inline src/client/actions/search_jobs.js]: :fire:

File: src/client/reducers/search_jobs_reducer.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    case FETCH_STUDENT_JOBS:
+        return {
+            ...state,
+            jobsRequest: {
+                status: action.status,
+                error: action.error,
+                response: action.response
+            }
+        };
+    case FETCH_SELECTED_JOBS:
```
Comment (des-des) [inline src/client/reducers/search_jobs_reducer.js]: Okay, so you only need one reducer.

Ie you only need `FETCH_STUDENT_JOBS`, then you can have more than one action that uses this.

File: database/database_queries.js (hunk 1 above)
Comment (des-des) [inline database/database_queries.js]:
```js
(callback) => {
  someAsyncFunc(args, (err, res) => {
    if (err) return callback(err)
    callback(null, res)
  }
}
```

is identical to 
```js
(callback) => {
  someAsyncFunc(args, callback)
}
```

File: src/client/actions/search_jobs.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    dispatch({
+        type: FETCH_SELECTED_JOBS,
```
Comment (des-des) [inline src/client/actions/search_jobs.js]: You can use the other action type for this:

`FETCH_STUDENT_JOBS`

File: src/client/components/pages/SearchJobs.js [hunk 9]
```diff
@@ -1,28 +1,168 @@
...
                         Browse Jobs
                     </label>
                     <input
-                        className="form__field-input"
+                        className="form-control"
                         id="Browse Jobs"
+                        type="text"
                         placeholder="Browse Jobs"
+                        list="jobs"
+                        onChange={this._onChange}
```
Comment (des-des) [inline src/client/components/pages/SearchJobs.js]: You do not need to use underscores in your function names.

Also, onChange is not good. onJobsSearchChange or something would be better

File: src/client/components/pages/SearchJobs.js [hunk 10]
```diff
@@ -1,28 +1,168 @@
...
+                        value={this.props.term}
```
Comment (des-des) [inline src/client/components/pages/SearchJobs.js]: term is a bad name, I do not know what it means

File: src/server/server.js [hunk 11]
```diff
@@ -7,40 +7,41 @@ const blipp = require('blipp');
...
 env('./config.env');
 
 const server = new Hapi.Server();
 
 const PORT = process.env.PORT || 4000;
 
 server.connection({
-    port: PORT,
+    port: PORT
     /*tls: process.env.NODE_ENV !== 'production' && {
```
Comment (des-des) [inline src/server/server.js]: Okay just delete all this, you can add it in later if you want it

File: src/client/actions/search_jobs.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+        .catch(err => {
+            dispatch({
+                type: FETCH_SELECTED_JOBS,
+                status: 'error',
+                error: err
+            });
+        });
+};
+
+export const setTerm = term => dispatch => {
```
Comment (des-des) [inline src/client/actions/search_jobs.js]: I do not know what `term` means

File: src/client/actions/search_jobs.js (hunk 6 above)
Comment (majakudlicka) [inline src/client/actions/search_jobs.js]: Done

File: src/client/actions/search_jobs.js (hunk 8 above)
Comment (majakudlicka) [inline src/client/actions/search_jobs.js]: I dont understand, these are two different API calls, how can we handle them with a single action?

File: src/client/actions/search_jobs.js (hunk 12 above)
Comment (majakudlicka) [inline src/client/actions/search_jobs.js]: Renamed to searchTerm. It is the term that user searches for when they look for job

File: src/client/reducers/search_jobs_reducer.js (hunk 7 above)
Comment (majakudlicka) [inline src/client/reducers/search_jobs_reducer.js]: Done

File: src/client/actions/search_jobs.js (hunk 8 above)
Comment (majakudlicka) [inline src/client/actions/search_jobs.js]: OK I think I got it, done

File: src/client/components/pages/SearchJobs.js (hunk 9 above)
Comment (majakudlicka) [inline src/client/components/pages/SearchJobs.js]: Done

File: src/client/components/pages/SearchJobs.js (hunk 10 above)
Comment (majakudlicka) [inline src/client/components/pages/SearchJobs.js]: Renamed

File: src/server/server.js (hunk 11 above)
Comment (majakudlicka) [inline src/server/server.js]: Removing this line breaks the proxy :/

**Pull request merged**